src/services/expenses.service.js

```diff
@@ -0,0 +1,69 @@
+const { data } = require('../data');
+
+const getAllExpenses = () => {
+  return data.expenses;
+};
+
+const getById = (id) => {
+  return data.expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  data.expenses = data.expenses.filter((exp) => exp.userId === parseInt(id));
```

Comment:
The function 'filterExpensesById' is directly modifying the 'data.expenses' array. This could lead to unexpected behavior if other parts of the code are relying on the original 'data.expenses' array. Consider returning a new array instead of modifying the existing one.